solidity/test/ValidatorAnnounce.t.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "../contracts/mock/MockMailbox.sol";
+import "../contracts/ValidatorAnnounce.sol";
+import {TypeCasts} from "../contracts/libs/TypeCasts.sol";
+
+contract ValidatorAnnounceTest is Test {
+    using TypeCasts for address;
+
+    // TODO: dedup
+    event ValidatorAnnouncement(
+        address indexed validator,
+        string storageLocation
+    );
+
+    MockMailbox mailbox;
+    uint32 localDomain = 1;
+    ValidatorAnnounce valAnnounce;
+
+    function setUp() public {
+        mailbox = new MockMailbox(localDomain);
+        valAnnounce = new ValidatorAnnounce(address(mailbox));
+    }
+
+    function announce(uint256 privateKey, string memory storageLocation)
+        internal
+    {
+        address validator = vm.addr(privateKey);
+        bytes32 domainHash = keccak256(
+            abi.encodePacked(
+                localDomain,
+                address(mailbox).addressToBytes32(),
+                "HYPERLANE"
+            )
+        );
+        bytes32 digest = ECDSA.toEthSignedMessageHash(
+            keccak256(abi.encodePacked(domainHash, storageLocation))
+        );
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(privateKey, digest);
+        bytes memory signature = abi.encodePacked(r, s, v);
+
+        valAnnounce.announce(validator, storageLocation, signature);
+    }
+
+    function assertEqAddrArr(address[] memory a, address[] memory b) internal {
+        bytes memory encodedA = abi.encodePacked(a);
+        bytes memory encodedB = abi.encodePacked(b);
+        assertEq(encodedA, encodedB);
+    }
+
+    function assertEqStrArrArr(string[][] memory a, string[][] memory b)
+        internal
+    {
+        assertEq(a.length, b.length);
+        bytes memory encodedA = abi.encode(a);
+        bytes memory encodedB = abi.encode(b);
+        assertEq(encodedA, encodedB);
```

Comment:
abi encode should encode the lengths no?

Reply:
Yeah I think so I was just being extra safe, will remove